File: src/main/golang/algorithm/sort/SelectionSort.go
package sort

func SelectionSort(arr []int) []int {

	for i := 0; i < len(arr) - 1; i++ {
		minIdx := i
		for j := i + 1; j < len(arr); j++ {
			if (arr[minIdx] > arr[j]) {
				minIdx = j
			}
		}
		arr[i], arr[minIdx] = arr[minIdx], arr[i]
	}

	return arr
}

File: src/main/java/algorithm/sort/SelectionSort.java
package algorithm.sort;

public class SelectionSort {

    public int[] selectionSort(int[] arr) {
        for (int i = 0; i < arr.length - 1; i++) {
            int minIdx = i;
            for (int j = i + 1; j < arr.length; j++) {
                if (arr[minIdx] > arr[j]) {
                    minIdx = j;
                }
            }
            swap(arr, i, minIdx);
        }

        return arr;
    }

    private void swap(int[] arr, int i, int minIdx) {
        int tmp = arr[i];
        arr[i] = arr[minIdx];
        arr[minIdx] = tmp;
    }

}
